src/llmcompressor/modifiers/smoothquant/base.py

```diff
@@ -329,4 +337,9 @@ def _calculate_smoothing_scales(
             1 - self.smoothing_strength
         )
         scales = torch.where(weight_scales > 0.0, scales, activation_scales)
+
+        for layer in balance_layers:
```

Comment:
Why can't the layer be re-offloaded on line 330? As currently written, there is a time where `balance_layers` are onloaded, which may lead to oom issues. The layer only needs to be onloaded while `layer.weight` is being used to calculate `weight_scales`

Reply:
It can. An open question I have is if we need to offload. See PR description

Reply:
https://github.com/vllm-project/llm-compressor/pull/978#issuecomment-2546666709